tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -184,3 +184,201 @@ def get_resource_params(self, m, r):
         # Docs are wrong :-(
         # https://cloud.google.com/compute/docs/reference/rest/v1/snapshots/delete
         return {'project': project, 'snapshot': snapshot_id}
+
+
+@resources.register('instance-group-manager')
+class InstanceGroupManager(QueryResourceManager):
+    """GCP resource: https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers
+    """
+    class resource_type(TypeInfo):
+        service = 'compute'
+        version = 'v1'
+        component = 'instanceGroupManagers'
+        enum_spec = ('aggregatedList', 'items.*.instanceGroupManagers[]', None)
+        scope = 'project'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_command(
+                'get', {'project': resource_info['project_id'],
+                        'zone': resource_info['location'],
+                        'instanceGroupManager': resource_info['instance_group_manager_name']})
+
+
+@InstanceGroupManager.action_registry.register('delete')
+class InstanceGroupManagerDelete(MethodAction):
+    """`Deletes <https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers
+    /delete>`_ an instance group manager
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: instance-group-manager-delete
+            resource: gcp.instance-group-manager
+            filters:
+              - type: value
+                key: name
+                op: eq
+                value: test-instance-group
+            actions:
+              - delete
+    """
+
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+    path_param_re = re.compile('.*?/projects/(.*?)/zones/(.*?)/instanceGroupManagers/(.*)')
+
+    def get_resource_params(self, m, r):
+        project, zone, name = self.path_param_re.match(r['selfLink']).groups()
+        return {'project': project, 'zone': zone, 'instanceGroupManager': name}
+
+
+@InstanceGroupManager.action_registry.register('set')
+class InstanceGroupManagerSet(MethodAction):
+    """
+    `Patches <https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers
+    /patch>`_ configuration parameters for an instance group.
+
+    The `instanceTemplate` represents an URL of the instance template that is specified for
+    this managed instance group.
+
+    The `autoHealingPolicies[].healthCheck` represents an URL for the health check that
```

Comment:
I left that as list intentionally - same as in API - to prevent breakage of policies in case GCP allows more than one health check in the future (I guess otherwise they hadn't done that array if they would have been sure than there will always be one and only one health check).
Wrt existing policies. We can always use filter to guarantee appliance of action only on those groups that have no health check yet.

Reply:
Any updates on this?